vms/platformvm/txs/mempool/mempool.go

```diff
@@ -306,11 +310,7 @@ func (m *mempool) deregister(tx *txs.Tx) {
 	m.consumedUTXOs.Difference(inputs)
 }
 
-func (m *mempool) RequestBuildBlock(emptyBlockPermitted bool) {
-	if !emptyBlockPermitted && !m.HasTxs() {
-		return
-	}
```

Comment:
Removing this check is what I was worried about. I think the explicit override is less error-prone- we don't want to accidentally build empty blocks

Reply:
Fixed

Reply:
I think this is still a concern, this PR accounts for not calling it in `Add` when there are no txs added but this method is also consumed outside of the `Mempool` interface

Reply:
I don't really see where the concern is

Reply:
This PR does not guard against sending extra `common.PendingTxs` messages if the mempool is empty in `RequestBuildBlock. Since it is an exported method on the `Mempool` interface, I believe it should guard against that. This is consistent to how the mempool is implemented in the AVM https://github.com/ava-labs/avalanchego/blob/dev/vms/avm/txs/mempool/mempool.go#L202-L204

With the way this PR is implemented, any consumers of the `Mempool` interface can call `RequestBuildBlock` un-restricted. When reviewing future PRs that interact with the mempool, we'll need to keep in mind that there is no guard for the P-Chain mempool. With an explicit boolean that overrides the check, reviewers can easily tell that the code is correct and does not send extra `common.PendingTxs` messages to the engine.

https://github.com/ava-labs/avalanchego/blob/simplify_block_build_request/vms/platformvm/txs/mempool/mempool.go#L243-L246
```golang
	// notify engine only if we are ready to build a block
	if addedTx {
		m.RequestBuildBlock()
	}
```

I'd like to move forward with the base PR for these reasons (consistency + more obviously correct code). If you disagree, let's discuss offline.

Reply:
I really don't think https://github.com/ava-labs/avalanchego/pull/2333 it's more obviously correct. I think the two alternatives are equally correct.
I would not speculate on what option is easier to inspect and instead wait for other reviewers to evaluate the proposals.